realm/realm-library/src/main/java/io/realm/SetValueOperator.java

```diff
@@ -533,6 +536,82 @@ boolean removeAllInternal(Collection<?> c) {
     }
 }
 
+/**
+ * TODO
+ */
+class UUIDOperator extends SetValueOperator<UUID> {
+
+    public UUIDOperator(BaseRealm baseRealm, OsSet osSet, Class<UUID> valueClass) {
+        super(baseRealm, osSet, valueClass);
+    }
+
+    @Override
+    boolean add(@Nullable UUID value) {
+        return osSet.add(value);
+    }
+
+    @Override
+    boolean contains(@Nullable Object o) {
+        UUID value;
+        if (o == null) {
+            value = null;
+        } else {
+            value = (UUID) o;
```

Comment:
Is this cast safe. I though I checked up on it in one of the other PRs, but smells like you could actually get called with arbitrary types all the way through to here. 

Similar on the other operations.

Reply:
Fixed, also other operations.